bin/build_supp.sh

```diff
@@ -177,6 +195,11 @@ function buildninja(){
   runcmd "wget https://github.com/ninja-build/ninja/archive/refs/tags/v${_version}.tar.gz"
   runcmd "tar -xzf v${_version}.tar.gz"
   runcmd "cd ninja-$_version"
+  _patch_file="$thisdir/patches/ninja-nprocs-v${_version}.patch"
+  if [ -r $_patch_file ]; then
+    runcmd   "cp $_patch_file $_builddir"
+    runcmdin "patch -p1" "$_patch_file"
```

Comment:
We might want to use `--merge` for the the `patch` command, as it not only says when the applying the patch failed, but also leaves `<<<<<` markers for the failed hunks in the target files.  That should greatly help debugging patches.